crates/pallet-bridges-initializer-currency-swap/src/upgrade_init.rs

```diff
@@ -0,0 +1,33 @@
+//! Initialization of the bridge pot accounts on runtime upgrade.
+
+use frame_support::pallet_prelude::*;
+
+use crate::Pallet;
+
+/// Initialize the bridges pot accounts.
+pub fn on_runtime_upgrade<T: crate::Config>() -> Weight {
+    match Pallet::<T>::initialize() {
```

Comment:
Here we need to check some sort of condition and only perform the init if necessary. I suggest that we add an explicit boolean (`has_initialized`) to the state to make things easier.

Reply:
https://github.com/humanode-network/humanode/pull/716/commits/a80bb29f3e19794ce39bbe7b91bc037249604db6